politeiawww/api/cms/v1/v1.go

```diff
@@ -433,3 +446,59 @@ type PaymentInformation struct {
 	AmountReceived  dcrutil.Amount `json:"amountreceived"`
 	Status          PaymentStatusT `json:"status"`
 }
+
+// User represents a CMS user. It contains the standard politeiawww user
+// fields as well as CMS specific user fields.
+type User struct {
+	ID                              string             `json:"id"`
+	Email                           string             `json:"email"`
+	Username                        string             `json:"username"`
+	Admin                           bool               `json:"isadmin"`
+	Identities                      []www.UserIdentity `json:"identities"`
+	LastLoginTime                   int64              `json:"lastlogintime"`
+	FailedLoginAttempts             uint64             `json:"failedloginattempts"`
+	Deactivated                     bool               `json:"isdeactivated"`
+	Locked                          bool               `json:"islocked"`
+	EmailNotifications              uint64             `json:"emailnotifications"` // Notify the user via emails
+	NewUserVerificationToken        []byte             `json:"newuserverificationtoken"`
+	NewUserVerificationExpiry       int64              `json:"newuserverificationexpiry"`
+	UpdateKeyVerificationToken      []byte             `json:"updatekeyverificationtoken"`
+	UpdateKeyVerificationExpiry     int64              `json:"updatekeyverificationexpiry"`
+	ResetPasswordVerificationToken  []byte             `json:"resetpasswordverificationtoken"`
+	ResetPasswordVerificationExpiry int64              `json:"resetpasswordverificationexpiry"`
+
+	// CMS Information
+	Domain             DomainTypeT     `json:"domain"` // Contractor domain
+	GitHubName         string          `json:"githubname"`
+	MatrixName         string          `json:"matrixname"`
+	ContractorType     ContractorTypeT `json:"contractortype"`
+	ContractorName     string          `json:"contractorname"`
+	ContractorLocation string          `json:"contractorlocation"`
+	ContractorContact  string          `json:"contractorcontact"`
+	SupervisorUserID   string          `json:"supervisoruserid"`
+}
+
+// UserDetails fetches a cms user's details by their id.
+type UserDetails struct {
+	UserID string `json:"userid"` // User id
+}
+
+// UserDetailsReply returns a cms user's details.
+type UserDetailsReply struct {
+	User User `json:"user"`
+}
+
+// EditUser edits a user's preferences.
+type EditUser struct {
+	Domain             DomainTypeT     `json:"domain"` // Contractor domain
```

Comment:
If a field is optional for the request you should add `omitemtpy` to the json tag.  I'm guessing that all of these fields are optional for the request, meaning you can edit just one at a time if you want to and aren't required to include all of them.